File: ios/Releaf/DesignSystem/CaptureMode.swift
/*
 * CaptureMode.swift
 *
 * The 7 capture flavors Releaf supports. Lives in the design system
 * because it's intrinsically a UI concept — the tab bar, quick-capture
 * sheet, and page-detail sections all render off it.
 *
 * Shape mirrors the Inkcreate mobile DS (title / subtitle / systemIcon).
 * `label` / `systemImage` are kept as back-compat aliases for older
 * call sites that haven't been migrated yet.
 */

import SwiftUI

public enum CaptureMode: String, CaseIterable, Identifiable, Sendable {
    case overview
    case photos
    case voice
    case todo
    case scans
    case contacts
    case location

    public var id: String { rawValue }

    public var title: String {
        switch self {
        case .overview: return "Overview"
        case .photos:   return "Photos"
        case .voice:    return "Voice note"
        case .todo:     return "To-do"
        case .scans:    return "Scan document"
        case .contacts: return "Contact"
        case .location: return "Location"
        }
    }

    public var subtitle: String {
        switch self {
        case .overview: return "All sections at a glance"
        case .photos:   return "Camera or upload"
        case .voice:    return "Record audio"
        case .todo:     return "Quick checklist item"
        case .scans:    return "Capture a document page"
        case .contacts: return "Phone, email, website"
        case .location: return "Tag current GPS"
        }
    }

    public var systemIcon: String {
        switch self {
        case .overview: return "square.grid.2x2"
        case .photos:   return "camera"
        case .voice:    return "mic"
        case .todo:     return "checklist"
        case .scans:    return "doc.viewfinder"
        case .contacts: return "person.crop.circle"
        case .location: return "mappin.and.ellipse"
        }
    }

    // MARK: - Back-compat aliases
    //
    // Older screens (pre-DS port) used `.label` / `.systemImage`. Kept as
    // thin shims so nothing breaks mid-port; remove once all call sites
    // are on the new API.

    public var label: String { title }
    public var systemImage: String { systemIcon }
}

File: android/app/src/main/java/app/releaf/mobile/ui/components/CaptureMode.kt
/*
 * CaptureMode.kt
 *
 * The 7 capture flavors Releaf supports. Lives in the design system
 * because it's intrinsically a UI concept — the tab bar, quick-capture
 * sheet, and page-detail sections all render off it.
 *
 * Shape mirrors the Inkcreate mobile DS (title / subtitle / icon).
 * `label` is kept as a back-compat alias for older call sites.
 */

package app.releaf.mobile.ui.components

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CameraAlt
import androidx.compose.material.icons.filled.Checklist
import androidx.compose.material.icons.filled.DocumentScanner
import androidx.compose.material.icons.filled.Eco
import androidx.compose.material.icons.filled.LocationOn
import androidx.compose.material.icons.filled.Mic
import androidx.compose.material.icons.filled.PersonOutline
import androidx.compose.ui.graphics.vector.ImageVector

enum class CaptureMode(
    val title: String,
    val subtitle: String,
    val icon: ImageVector,
) {
    // Overview is the page's "everything at a glance" tab; the leaf
    // is Releaf's brand glyph (see BRAND_BRIEF.md), so the overview
    // tab carries it. Active state fills the leaf onto the
    // accent-palette square; inactive state shows the same glyph in
    // the default text color.
    Overview("Overview",       "All sections at a glance", Icons.Filled.Eco),
    Photos  ("Photos",         "Camera or upload",         Icons.Filled.CameraAlt),
    Voice   ("Voice note",     "Record audio",             Icons.Filled.Mic),
    Todo    ("To-do",          "Quick checklist item",     Icons.Filled.Checklist),
    Scans   ("Scan document",  "Capture a document page",  Icons.Filled.DocumentScanner),
    Contacts("Contact",        "Phone, email, website",    Icons.Filled.PersonOutline),
    Location("Location",       "Tag current GPS",          Icons.Filled.LocationOn);

    /** Back-compat alias for pre-port call sites. */
    val label: String get() = title
}
